.NET/Microsoft.Recognizers.Text.NumberWithUnit/Extractors/NumberWithUnitExtractor.cs

```diff
@@ -259,6 +259,28 @@ public List<ExtractResult> Extract(string source)
                         /* Relative position will be used in Parser */
                         number.Start = start - er.Start;
                         er.Data = number;
+
+                        //Special treatment, handle cases like '2:00 pm', '00 pm' is not dimension

```

Comment:
This could a be a method to make the code cleaner and easy to modify. Also, the solution could be more generic to allow handling other potentially ambiguous terms.